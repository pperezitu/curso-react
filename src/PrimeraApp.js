import React from "react";
import PropTypes from 'prop-types';
//import React, { Fragment } from "react";

const PrimeraApp = ( { saludo, subtitulo } )=> {
//const PrimeraApp = ( props )=> {

    // if(!saludo) {
    //     throw new Error('El suludo es necesario');
    // }

    // const saludo = "Hola Mundo";
    // console.log(props)    
    return (
        <>
            <h1>{saludo}</h1>
            {/* <h1> {props.saludo} </h1> */}
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Primera App con React'
}

export default PrimeraApp;
