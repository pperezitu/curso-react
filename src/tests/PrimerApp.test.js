import { React } from 'react';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en primera App', () => {
    // test('debe mostrar mensaje "Hola, Soy Goku!!"', () => {
        
    //     const saludo = "Hola, Soy Goku!!";
    //     const { getByText } = render(<PrimeraApp  saludo={ saludo } />);

    //     expect(getByText(saludo)).toBeInTheDocument();

    // });

    test('debe mostrar <PrimeraApp/> correctamente', () => {

        const saludo = "Hola, Soy Goku!!";
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
    });
    
    
})