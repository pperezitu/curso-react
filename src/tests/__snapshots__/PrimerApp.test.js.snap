// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera App debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku!!
  </h1>
  <p>
    Primera App con React
  </p>
</Fragment>
`;
